src/dwarf/resolver.rs

```diff
@@ -93,6 +97,29 @@ impl<'mmap> DebugSyms<'mmap> {
         }
     }
 
+    /// Find a symbol by address.
+    fn find_by_addr<'slf>(
+        &'slf self,
+        addr: Addr,
+    ) -> impl Iterator<Item = &'slf DWSymInfo<'mmap>> + Clone + 'slf {
+        let idx =
+            if let Some(idx) = find_match_or_lower_bound_by_key(&self.syms, addr, |sym| sym.addr) {
+                idx
+            } else {
+                return Either::A([].into_iter())
+            };
+
+        let syms = self.syms[idx..].iter().take_while(move |sym| {
+            if sym.size == 0 {
+                sym.addr == addr
+            } else {
+                (sym.addr..sym.addr + sym.size).contains(&addr)
+            }
```

Comment:
this logic seems common and kind of distracting, maybe have a helper to check if symbol overlaps the address?

Reply:
Don't know if it makes sense to have a "generic" utility function, but I added two `contains` functions to the individual types in question.